subsys/caf/modules/ble_state.c

```diff
@@ -189,19 +201,18 @@ static void connected(struct bt_conn *conn, uint8_t error)
 	if (IS_ENABLED(CONFIG_BT_PERIPHERAL) &&
 	    (info.role == BT_CONN_ROLE_PERIPHERAL)) {
 		struct bond_find_data bond_find_data = {
-			.peer_id = 0,
-			.peer_count = 0,
+			.peer_address = bt_conn_get_dst(conn),
+			.peer_bonded = false,
+			.bond_cnt = 0
 		};
-		bt_foreach_bond(info.id, bond_find, &bond_find_data);
-
-		LOG_INF("Identity %u has %u bonds", info.id,
-			bond_find_data.peer_count);
-		if ((bond_find_data.peer_count > 0) &&
-		    bt_addr_le_cmp(bt_conn_get_dst(conn),
-				   &bond_find_data.peer_address)) {
-			bt_addr_le_to_str(&bond_find_data.peer_address, addr_str,
-					sizeof(addr_str));
-			LOG_INF("Already bonded to %s", addr_str);
```

Comment:
I wonder why we disconnected in that case. We no longer do that which seems logical, but why was the disconnection present in first place?

Reply:
We disconnected from a connection if the used local identity was already bonded with another device (to prevent bonding with 2 peers on the same local identity).

Reply:
We still keep similar behavior here, but we allow for more than one bond per Bluetooth local identity.

Reply:
Oh, okay I read the conditions wrongly. Now it makes sense. Never mind.